zkevm-circuits/src/super_circuit.rs

```diff
@@ -250,6 +264,31 @@ impl<F: Field> SubCircuitConfig<F> for SuperCircuitConfig<F> {
             },
         );
 
+        // constraint chronological/by address rwtable row fingerprint must be the same in first
+        // chunk first row.
+        meta.create_gate(
+            "chronological rwtable row fingerprint == by address rwtable row fingerprint",
```

Comment:
This seems already guaranteed by the `RootCircuit` right (by initializing them both as `1`)?

Reply:
There are 2 terms `row_fingerprint` and `acc_fingerprint`

- acc_fingerprint: start from 1
- row_fingerprint: row level rlc value by `alpha/gamma` challenge

by current design: `row_fingerprint(0) v acc_fingerprint_rotation(-1) = acc_fingerprint_rotation(-1)`

Root circuit only guarantee `acc_fingerprint` at offset 0 as `1`. However, root circuit do not know what the first `row_fingerprints` looks like (In our case, first row first chunk in rw_table is `Rw::Start`). This value can only being known in super circuit thats's why this gates is added. 

Besides, `BeginChunk` virtual step will also lookup `Rw::Start`. So

- by address rw table first row must be Start -> guaranteed by BeginChunk lookup 
- chronologically rw_table first row must equal with state_circuit -> guaranteed by this new gate

Thanks for the reminding and I will add more code comments to explain this :)

Reply:
I see I see, now I get it, thanks for clarifying!